bless/ssh/certificate_authorities/rsa_certificate_authority.py

```diff
@@ -4,27 +4,31 @@
     :license: Apache, see LICENSE for more details.
 """
 from bless.ssh.certificate_authorities.ssh_certificate_authority import \
-    SSHCertificateAuthority
+    SSHCertificateAuthority, SSHCertificateSignetureKeyType
 from bless.ssh.protocol.ssh_protocol import pack_ssh_mpint, pack_ssh_string
-from bless.ssh.public_keys.ssh_public_key import SSHPublicKeyType
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import hashes
 from cryptography.hazmat.primitives.asymmetric import padding
 from cryptography.hazmat.primitives.serialization import load_pem_private_key
 
 
 class RSACertificateAuthority(SSHCertificateAuthority):
-    def __init__(self, pem_private_key, private_key_password=None):
+    def __init__(self, pem_private_key, private_key_password=None, cert_type="sha2"):
         """
         RSA Certificate Authority used to sign certificates.
         :param pem_private_key: PEM formatted RSA Private Key.  It should be encrypted with a
         password, but that is not required.
         :param private_key_password: Password to decrypt the PEM RSA Private Key, if it is
         encrypted.  Which it should be.
         """
-        super(SSHCertificateAuthority, self).__init__()
-        self.public_key_type = SSHPublicKeyType.RSA
+        super().__init__()
 
+        if cert_type == "sha1":
+            self.public_key_type = SSHCertificateSignetureKeyType.RSA
+            self.algo = hashes.SHA1()
+        else:
+            self.public_key_type = SSHCertificateSignetureKeyType.RSA_SHA2
```

Comment:
the PR is in a working state - e2e tested, unit tests passing

for line 30 here, RSA_SHA2 key type does not seem to be used, because line 49 has already hard coded the key type to be RSA. However if changing line 30 to be 'RSA', e2e test would break with warning:

"WARN[0005] Error adding certificate to ssh-agent, is your ssh-agent running?  error="Could not add key/certificate to SSH_AGENT_SOCK: agent: failure""

Still investigating on where this is referenced.

Reply:
Ok so [this public_key_type](https://github.com/lyft/bless/blob/lyft-host-certificate-lambda_2/bless/ssh/certificate_authorities/ssh_certificate_authority.py#L40) is used to append to the signed cert - that's why keeping it as RSA_SHA2 is necessary.